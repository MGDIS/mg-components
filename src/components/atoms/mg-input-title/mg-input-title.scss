/**
 * Not scoped into `mg-input-title` because it could be used outside of an input
 * (ex: All input with a * are required)
 */
.is-asterisk {
  color: hsl(var(--color-danger));
}

legend.mg-input-title {
  padding-left: 0;
  padding-right: 0;
}
