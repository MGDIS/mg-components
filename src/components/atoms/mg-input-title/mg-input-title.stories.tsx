import { h } from "@stencil/core";

export default {
  component: 'mg-input-title',
  title: 'Atoms/mg-input-title'
};

const Template = args => {
  // Extract slot so it won't be render as an attribute
  const slot = args.slot;
  delete args.slot;
  // return element
  return <mg-input-title {...args}>{slot}</mg-input-title>
};

export const MgInputTitle = Template.bind({});
MgInputTitle.args = {
  slot: 'Label',
  identifier: 'identifier',
  required: true,
};
