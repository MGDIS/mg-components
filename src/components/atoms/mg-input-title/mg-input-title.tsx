import { Component, h, Prop, State } from '@stencil/core';

@Component({
  tag: 'mg-input-title',
  styleUrl: 'mg-input-title.scss',
  scoped: true,
})
export class MgInputTitle {
  /**
   * Label input id
   */
  @Prop() identifier!: string;

  /**
   * If input is required an asterisk is added at the end of the label
   */
  @Prop() required: boolean;

  /**
   * Switch from label to fieldset sementic
   */
  @Prop() isLegend: boolean = false;

  /**
   * Component parent tagname
   */
  @State() tagName: string = 'label';

  private getTagName() {
    this.tagName = this.isLegend ? 'legend' : 'label';
  }

  /*************
   * Lifecycle *
   *************/

  componentWillLoad() {
    // Init tag name
    this.getTagName();
  }

  // \u00A0 represent a &nbsp;
  render() {
    const TagName = this.tagName;

    return (
      <TagName class="mg-input-title" htmlFor={this.identifier}>
        <slot></slot>
        {this.required && [`\u00A0`, <span class="is-asterisk">*</span>]}
      </TagName>
    );
  }
}
