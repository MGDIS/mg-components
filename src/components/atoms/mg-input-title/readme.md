This atom refers to thes [PDA9-64](https://jira.mgdis.fr/browse/PDA9-64), [PDA9-723](https://jira.mgdis.fr/browse/PDA9-723).

<!-- Auto Generated Below -->


## Properties

| Property                  | Attribute    | Description                                                       | Type      | Default     |
| ------------------------- | ------------ | ----------------------------------------------------------------- | --------- | ----------- |
| `identifier` _(required)_ | `identifier` | Label input id                                                    | `string`  | `undefined` |
| `isLegend`                | `is-legend`  | Switch from label to fieldset sementic                            | `boolean` | `false`     |
| `required`                | `required`   | If input is required an asterisk is added at the end of the label | `boolean` | `undefined` |


## Dependencies

### Used by

 - [mg-input-checkbox](../../molecules/inputs/mg-input-checkbox)
 - [mg-input-date](../../molecules/inputs/mg-input-date)
 - [mg-input-numeric](../../molecules/inputs/mg-input-numeric)
 - [mg-input-password](../../molecules/inputs/mg-input-password)
 - [mg-input-radio](../../molecules/inputs/mg-input-radio)
 - [mg-input-select](../../molecules/inputs/mg-input-select)
 - [mg-input-text](../../molecules/inputs/mg-input-text)
 - [mg-input-textarea](../../molecules/inputs/mg-input-textarea)

### Graph
```mermaid
graph TD;
  mg-input-checkbox --> mg-input-title
  mg-input-date --> mg-input-title
  mg-input-numeric --> mg-input-title
  mg-input-password --> mg-input-title
  mg-input-radio --> mg-input-title
  mg-input-select --> mg-input-title
  mg-input-text --> mg-input-title
  mg-input-textarea --> mg-input-title
  style mg-input-title fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
