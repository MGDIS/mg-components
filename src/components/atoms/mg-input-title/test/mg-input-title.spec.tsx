import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgInputTitle } from '../mg-input-title';

const getPage = args =>
  newSpecPage({
    components: [MgInputTitle],
    template: () => <mg-input-title {...args}>mg-input-title</mg-input-title>,
  });

describe('mg-input-title', () => {
  test.each([
    { required: false, identifier: 'identifier' },
    { required: true, identifier: 'identifier' },
    { isLegend: true, identifier: 'identifier' },
  ])('Should render label with args %s', async args => {
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });

  test.each(['', ' ', undefined])('Should throw error, case invalid identifier prop', async identifier => {
    expect.assertions(1);
    try {
      await getPage({ identifier });
    } catch (err) {
      expect(err.message).toContain('<mg-input-title> prop "identifier" is required.');
    }
  });
});
