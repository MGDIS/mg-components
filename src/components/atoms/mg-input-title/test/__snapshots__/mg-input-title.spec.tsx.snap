// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-input-title Should render label with args { isLegend: true, identifier: 'identifier' } 1`] = `
<mg-input-title>
  <!---->
  <legend class="mg-input-title">
    mg-input-title
  </legend>
</mg-input-title>
`;

exports[`mg-input-title Should render label with args { required: false, identifier: 'identifier' } 1`] = `
<mg-input-title>
  <!---->
  <label class="mg-input-title" htmlfor="identifier">
    mg-input-title
  </label>
</mg-input-title>
`;

exports[`mg-input-title Should render label with args { required: true, identifier: 'identifier' } 1`] = `
<mg-input-title>
  <!---->
  <label class="mg-input-title" htmlfor="identifier">
    mg-input-title
    <span class="is-asterisk">
      *
    </span>
  </label>
</mg-input-title>
`;
